### PR TITLE
stop and remove docker container

### PR DESCRIPTION
**Pull request by 404fixer**
Closes: #708 

#### Describe the changes you've made
Earlier after stopping keploy, it used to stop the docker container but not remove it. Now, the container will be stopped and removed.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Please let us know if any test cases are added

Please describe the tests(if any). Provide instructions how its affecting the coverage.

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)
A clear and concise description of it.

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by 404fixer** [comment]
I have read the CLA Document and I hereby sign the CLA

File: pkg/hooks/launch.go [hunk 1]
```diff
@@ -202,6 +210,7 @@ func (h *Hook) processDockerEnv(appCmd, appContainer, appNetwork string) error {
 			case e := <-messages:
 				if e.Type == events.ContainerEventType && e.Action == "create" {
 					// Fetch container details by inspecting using container ID to check if container is created
+					h.dockerDetails.containerID = e.ID
```
Comment (charankamarapu) [inline pkg/hooks/launch.go:213]: Use setter and getter functions instead of assigning it directly

File: pkg/hooks/loader.go [hunk 2]
```diff
@@ -334,13 +342,48 @@ func (h *Hook) findAndCollectChildProcesses(parentPID string, pids *[]int) {
...
 }
 
+// Stop and Remove the docker container
+func (h *Hook) StopAndRemoveDockerContainer() {
+	dockerClient := h.idc
+	containerID := h.dockerDetails.containerID
+
+	container, err := dockerClient.ContainerInspect(context.Background(), containerID)
+    if err != nil {
+        fmt.Println(err)
```
Comment (charankamarapu) [inline pkg/hooks/loader.go:352]: Use logger instead of print statements

File: pkg/hooks/loader.go [hunk 3]
```diff
@@ -334,13 +342,48 @@ func (h *Hook) findAndCollectChildProcesses(parentPID string, pids *[]int) {
...
+        return
```
Comment (charankamarapu) [inline pkg/hooks/loader.go:353]: why is err not being returned..?

File: pkg/hooks/loader.go [hunk 4]
```diff
@@ -334,13 +342,48 @@ func (h *Hook) findAndCollectChildProcesses(parentPID string, pids *[]int) {
...
+    }
+
+	removeOptions := types.ContainerRemoveOptions{
+		RemoveVolumes: true,
+		Force:         true,
+	}
+
+    err = dockerClient.ContainerRemove(context.Background(), containerID, removeOptions)
+    if err != nil {
+        fmt.Println(err)
```
Comment (charankamarapu) [inline pkg/hooks/loader.go:371]: Use log statements

**Review by charankamarapu — COMMENTED** [review]
Please resolve the mentioned issues

File: pkg/hooks/loader.go [hunk 5]
```diff
@@ -342,47 +334,21 @@ func (h *Hook) findAndCollectChildProcesses(parentPID string, pids *[]int) {
...
 			return
 		}
-		
-		// Stop Docker Container and Remove it
-		h.StopAndRemoveDockerContainer()
+
+		// Stop Docker Container and Remove it if Keploy ran using docker
+		if len(h.idc.GetContainerID()) != 0  {
+			err := h.idc.StopAndRemoveDockerContainer()
+			if err != nil {
```
Comment (charankamarapu) [inline pkg/hooks/loader.go:348]: This should be an error log saying failed to stop container/ remove container , zap the error and suggest user to delete it manually

**Review by charankamarapu — COMMENTED** [review]
Everything else looks good to me. Please correct the logger statement

**Review by charankamarapu — APPROVED** [review]
Ship it!

**Review by re-Tick — APPROVED** [review]
LGTM

**Pull request merged**